src/foam/core/XML.js

```diff
@@ -0,0 +1,466 @@
+        Function:  function(o) { this.outputFunction(o); },
+        AbstractEnum: function(o) { },
+        FObject:   function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this)
+            return;
+          }
+
+          var clsName = o.cls_.id;
+          this.start("<object class='" + clsName + "'>")
+          // Iterate through properties and output
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputProperty_(o, ps[i]);
+          }
+          this.end('</object>');
+        },
+        Array: function(o) {
+          // Nested Objects and FObject Arrays Passed
+          for ( var i = 0 ; i < o.length ; i++ ) {
+            // Output 'value' tags for arrays containing non-FObject values
+            if ( !o[i].cls_) this.out("<value>");
+            this.output(o[i], this);
+            if ( !o[i].cls_ ) this.out("</value>");
+            if ( o.length - i != 1 ) this.nl().indent();
+          }
+        },
+        Object: function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this);
+          } else {
+            var oName = o.name;
+            this.start("<object name='" + oName + "'>");
+            if ( this.sortObjectKeys ) {
+              this.outputSortedObjectKeyValues_(o);
+            } else {
+              this.outputObjectKeyValues_(o);
+            }
+            this.end('</object>');
+          }
+        }
+      })
+    },
+
+    function stringify(o) {
+      // Root tags of objects for array of FObjects
+      if ( o instanceof Array ) {
+        this.start("<objects>");
+        this.output(o);
+        this.end("</objects>");
+      } else {
+        this.output(o);
+      }
+      var ret = this.buf_;
+      this.reset(); // reset to avoid retaining garbage
+      return ret;
+    },
+
+    {
+      name: 'createFObj',
+      code: foam.mmethod({
+        Object: function (o) {
+          // Create FObject
+          var className = o.className;
+          var obj = foam.lookup(className).create();
+          var props = o.children;
+
+          // Populate FObject with properties
+          for ( var propIndex = 0; propIndex < props.length; propIndex++ ) {
+
+            var currentProp = props[propIndex];
+            var prop = obj.cls_.getAxiomByName(currentProp.tagName);
+            var childName = currentProp.firstChild.localName;
+            // Specific case for array
+            if ( currentProp.className === 'Array' ) {
+              // Array of FObjects
+              if ( childName === 'object' ) {
+                var nestObjArray = Array.from(currentProp.childNodes);
+                prop.set(obj, this.createFObj(nestObjArray));
+              } else {
+                // Array of other objects with 'value' tag names
+                var arrayValue = (Array.from(currentProp.children)).map( function (x) { return x.innerHTML; });
```

Comment:
I assumed that when we set the property of an Object, there was an `adapt` method within PropertyInfo that would parse correctly from String and convert the correct type.